chai-jquery.js

```diff
@@ -228,4 +228,21 @@
       };
     }
   );
+
+  chai.Assertion.addMethod('focus', function () {
+    // workaround for Webkit-based browsers
+    // see https://github.com/ariya/phantomjs/issues/10427
+    var _jQuery_is = $.fn.is;
+    $.fn.is = function(s) {
+    if (s === ':focus') {
+      return this.get(0) === document.activeElement;
+    }
+      return _jQuery_is.apply(this, arguments);
+    };
+    
+    this.assert(
+        flag(this, 'object').is(':focus')
+      , 'expected #{this} to have focus'
+      , 'expected #{this} not to have focus');
+  });
```

Comment:
@T1st3 rather than monkey patching jquery itself, why don't you just take the logic out of this overload, and add that to the assertion.

For example, you could have:

```js
chai.Assertion.addMethod('focus', function () {
  this.assert(
      // Can't use `$().is(':focus')` because of certain webkit browsers
      // see https://github.com/ariya/phantomjs/issues/10427
      flag(this, 'object').get(0) === document.activeElement
    , 'expected #{this} to have focus'
    , 'expected #{this} not to have focus');
});
```

This way we won't be permanently changing the behaviour of jQuery when people use the focus assertion.